avocado/core/job.py

```diff
@@ -89,7 +89,9 @@ def __init__(self, args=None):
         if unique_id is None:
             unique_id = job_id.create_unique_job_id()
         self.unique_id = unique_id
-        self.logdir = None
+        self._setup_job_results()
+        if not (self.standalone or getattr(self.args, "dry_run", False)):
+            self._update_latest_link()
```

Comment:
Well I agree that everything important should happen inside the scope of the job, but I'd rather keep the `setup_job_results` as separate entity before we start doing anything else and not directly inside the `__init__`. The `__init__` instantiates the job, then users might want to check some things (network connection, ...) and then start the job. I'd see those again as part of the pre-tests setup phase with priority ordering. On the other hand we do have the `job_id` generated in `__init__` as well, so probably we we can say it's priority 0 and do it directly inside `__init__`.

As for the `update_latest_link`, I'd make it part of the `setup_job_results` as in my view it is it's part.